back/engines/commercial/idea_custom_fields/app/controllers/idea_custom_fields/web_api/v1/admin/idea_custom_fields_controller.rb

```diff
@@ -106,23 +106,36 @@ def create_field!(field_params, errors, page_temp_ids_to_ids_mapping, index)
         create_params['key'] = default_field.key
       end
       field = CustomField.new create_params.merge(resource: @custom_form)
-      SideFxCustomFieldService.new.before_create field, current_user
-      if field.save
-        page_temp_ids_to_ids_mapping[field_params[:temp_id]] = field.id if field_params[:temp_id]
-        SideFxCustomFieldService.new.after_create field, current_user
-        field
+
+      IdeaCustomFieldsService.new(@custom_form).validate_constraints_against_defaults(field)
+      if field.errors.errors.empty?
+        SideFxCustomFieldService.new.before_create field, current_user
+        if field.save
+          page_temp_ids_to_ids_mapping[field_params[:temp_id]] = field.id if field_params[:temp_id]
+          SideFxCustomFieldService.new.after_create field, current_user
+          field
+        else
```

Comment:
Got this duplicate error syntax, which is annoying. But every way I tried to combine it into one loop the response was always returning an empty error object instead of the response for a valid response